docs/src/user-guide/configuring/configuration-files-new.md

```diff
@@ -547,7 +549,7 @@ Here, the second configuration object only overrides the severity, so the final
 
 ### Configuring shared settings
 
-ESLint supports adding shared settings into configuration files. Plugins use `settings` to specify information that should be shared across all of its rules. You can add a `settings` object to a configuration object and it will be supplied to every rule being executed. This may be useful if you are adding custom rules and want them to have access to the same information. Here's an example:
+ESLint supports adding shared settings into configuration files. When you add a `settings` object to a configuration object, it is supplied to every rule executed by that configuration object. By convention, plugins namespace the settings they are interested in to avoid collisions with others. Plugins can use `settings` to specify the information that should be shared across all of their rules. This may be useful if you are adding custom rules and want them to have access to the same information. Here's an example:
```

Comment:
"it is supplied to every rule executed by that configuration object" might be misleading. It sounds like the `settings` object is only supplied to `rules: {}` from the same configuration object.

Reply:
@mdjermanovic 
what'd be correct, it's applied to all rules in the project?

Reply:
I think you can just drop “executed by that configuration object”.

Reply:
It's supplied to every rule when linting a file to which the configuration object applies, but those rules don't have to be enabled by that configuration object.

For example:

```js
module.exports = [
    {
        rules: {
            "my-plugin/my-rule": "error"
        }
    },
    {
        files: ["a.js"],
        settings: {
            data: "something"
        }
    }
];
```

When linting `a.js`, `my-plugin/my-rule` will be supplied with `{ data: "something" }` (as `context.settings`), although `my-plugin/my-rule` rule isn't enabled in the same configuration object where `settings` are defined.

Since this sentence only briefly describes what `settings` means, it could just say "it is supplied to every rule", as @nzakas suggested.

Reply:
thanks for the further clarification, implementing nicholas's suggestion!